src/documents/serialisers.py

```diff
@@ -681,12 +686,48 @@ def update(self, instance: Document, validated_data):
                             custom_field_instance.field,
                             doc_id,
                         )
+        if (
+            "remove_inbox_tags" in validated_data
+            and validated_data["remove_inbox_tags"]
+        ):
+            tag_ids_being_added = (
+                [
+                    tag.id
+                    for tag in validated_data["tags"]
+                    if tag not in instance.tags.all()
+                ]
+                if "tags" in validated_data
+                else []
+            )
+            inbox_tags_not_being_added = Tag.objects.filter(is_inbox_tag=True).exclude(
+                id__in=tag_ids_being_added,
+            )
+            if "tags" in validated_data:
+                validated_data["tags"] = [
+                    tag
+                    for tag in validated_data["tags"]
+                    if tag not in inbox_tags_not_being_added
+                ]
+            else:
+                validated_data["tags"] = [
+                    tag
+                    for tag in instance.tags.all()
+                    if tag not in inbox_tags_not_being_added
+                ]
         super().update(instance, validated_data)
         return instance
 
     def __init__(self, *args, **kwargs):
         self.truncate_content = kwargs.pop("truncate_content", False)
 
+        # return full permissions if we're doing a PATCH or PUT
```

Comment:
Yea, exactly it prevents "losing" the permissions on patch or put. They're not actually lost but if the returned document doesnt have them the frontend effectively would remove them and then if you hit save you would actually lose them. If that makes sense...

Reply:
This if youre curious, the frontend now patches in the values returned from the API after a save because the returned values might be different than what was sent (effectively what this PR does):

https://github.com/paperless-ngx/paperless-ngx/blob/6292de5c6034b3f7d04c3027d97bb679b617ea13/src-ui/src/app/components/document-detail/document-detail.component.ts#L629-L635